src/backend/gl/src/lib.rs

```diff
@@ -37,15 +39,91 @@ mod state;
 mod window;
 
 #[cfg(feature = "glutin")]
-pub use window::glutin::{config_context, Headless, Surface, Swapchain};
+pub use window::glutin::{Surface, Swapchain, Window};
+
+#[cfg(feature = "glutin")]
+pub struct Instance {
+    el: Arc<Mutex<glutin::EventsLoop>>,
+    context: glutin::Context,
+    pub(crate) quadvbo: Arc<Mutex<Option<u32>>>,
+    pub(crate) quadprogram: Arc<Mutex<Option<u32>>>,
+}
+
+#[cfg(feature = "glutin")]
+pub enum InstanceOrWindow {
+    Instance(Arc<Instance>),
+    Window(Starc<glutin::GlWindow>),
+}
+
+#[cfg(feature = "glutin")]
+impl Deref for InstanceOrWindow {
+    type Target = glutin::Context;
+    fn deref(&self) -> &Self::Target {
+        match self {
+            InstanceOrWindow::Instance(i) => &i.context,
+            InstanceOrWindow::Window(w) => w.context(),
+        }
+    }
+}
+
+#[cfg(feature = "glutin")]
+impl Instance {
+    pub fn create(_name: &str, _version: u32, el: Arc<Mutex<glutin::EventsLoop>>) -> Arc<Self> {
+        let cb = glutin::ContextBuilder::new();
+        let context = glutin::Context::new(cb, &el.lock().unwrap(), true).unwrap();
+
+        Arc::new(Instance {
+            el,
+            context,
+            quadvbo: Arc::new(Mutex::new(None)),
```

Comment:
why aren't we initializing those here?

Reply:
We haven't loaded the gl pointers yet nor done the stuff which is done when opening the device. I'm not actually sure what the opening the device stuff is done for so I avoided changing that code and just made the stuff get initialized when making a swap chain the first time.

Reply:
I guess we could make the physical device & open the device one in the instance then just return those. I think that's what I'll change it to do, and then we can avoid reloading the gl ptrs in drop.